src/Analysis/Ast/Impl/Analyzer/ModuleWalker.cs

```diff
@@ -83,9 +97,18 @@ public ModuleWalker(IServiceContainer services, IPythonModule module, PythonAst
             SymbolTable.EvaluateAll();
             SymbolTable.ReplacedByStubs.Clear();
             MergeStub();
+
+            if (_allReferencesCount == 1 && GlobalScope.Variables.TryGetVariable(AllVariableName, out var variable) && variable?.Value is IPythonCollection collection) {
```

Comment:
I guess @jakebailey and https://github.com/microsoft/python-language-server/issues/620 should be extending this.